### ISSUE TITLE
PSVita Android Game Ports with two START buttons

### ISSUE THREAD

**Issue by billabongbruno** [BlackSheepBoy69/HexFlow-Launcher-Unofficial-Custom#11]
Hey there.
Sorry for being a hassle again.

I figured I would open a new issue, seeing as this is unrelated to the enhancement we were discussing.

Today I installed This War of Mine with the DLC "Stories" and realized that it has two different START buttons (each redirecting to a different portion of the game).

By default, HexFlow Custom launches the top one and even if I press the bottom one, it has no effect in changing the game.

Is there any hidden option to enable a non-automatic-start on a per-game basis or an option to stop auto-loading when there is more than one START instance that I'm missing?

Sorry for bothering you again.

Thank you in advance.

Best regards,
Bruno.

**Comment by BlackSheepBoy69** [comment]
You are no hassle! I'll check this out on the weekend-

**Comment by billabongbruno** [comment]
> You are no hassle! I'll check this out on the weekend-

Well, I suppose I am, seeing as this is the second thing I ask of you. lol

Thank you for checking this out.

See you soon.

Bruno.

**Comment by BlackSheepBoy69** [comment]
The 2nd 'start button' is actually just a banner. I might say split the app into 2 apps, but unfortunately the game only has one binary (the .bin file inside ux0:app/TWOM00000 ). The ability for HexFlow to use banners would be really complicated and I wouldn't really know a clean way to implement it (it would get messy for Japanese games which often have sooo many banners)

**Comment by billabongbruno** [comment]
> The 2nd 'start button' is actually just a banner. I might say split the app into 2 apps, but unfortunately the game only has one binary (the .bin file inside ux0:app/TWOM00000 ) so it would be a lot more entailed, for example, having to mess with all Rinne's code and compile to 2 binaries.
> 
> The ability for HexFlow to use banners would be really complicated and I wouldn't really know a clean way to implement it. It might be better to post it as an issue on TWOM to say like "Hey, if I have the game 1 open and hit the banner, make it close the game and run game 2" instead of keep playing game 1 which it seems to do"
> 
> Just closing the issue for now because it seems more of an issue with TWOM (I don't know any other games that have this issue, standard procedure is to have a little in-game mini-menu inside that lets you pick the game)

I see.
That's unfortunate, but not a big issue.

I have come across more Android ports that have multiple banners, but they are all "Configurator" banners rather than other portions of the games themselves, so nothing really relevant. The only one that has another game baked into it seems to be TWOM.

Thank you for your reply, nevertheless.

Unrelated to HexFlow, do you know if there is anyway I can create a custom PSVita Bubble that sets its main banner to the content of TWOM's secondary banner, as sort of a workaround?

(Kind of like creating a clone bubble, but redirecting its shorcuts, which I believe pointed to something like "psla:stories" or something)

Is this possible or would it require editing the eboot.bin?

If not at all possible, I will probably keep TWOM's bubble visible, along with HexFlow Custom.


Thank you once more for all of your help and the time you took to reply.

Best regards,
Bruno.

**Comment by BlackSheepBoy69** [comment]
Actually I know a really easy way to solve this... 1 sec

**Comment by BlackSheepBoy69** [comment]
Nope... For whatever reason it seems to refuse to launch directly by a separate bubble. You can find my attempt here:

https://www.mediafire.com/file/ylga5a2om8fv0zt/TWOM_Stories_Redirector_FAILED.vpk/file

Note: The apptitle is "Wordle" because I was too lazy to change it.

I'm not super passionate about getting it fixed, if you are, please go to the This War Of Mine Github page and open an issue:

Title: LUA Direct Launch

Text something like: I was trying to make a little separate bubble that could launch straight to TWOM:Stories (so it could easily be accessed by HexFlow Launcher) using your Lua Player Plus. I tried this:
	    System.executeUri("psgm:play?titleid=TWOM00000&param=stories")
but it doesn't seem to work. Do you know what would be able to launch into TWOM:Stories?

Not part of the text to put in the issue, this is just me talking: I really did try that but it didn't work. For the wordle thing I just made a copy of the TWOM game, stuffed it into a Wordle VPK (since that's the only homebrew I know that primarily uses PSLA: launch. If you want to try out LUA code, just put it in HexFlow Launcher Custom right below the area that says:

	elseif (Controls.check(pad, SCE_CTRL_SELECT) and not Controls.check(oldpad, SCE_CTRL_SELECT)) then

(you may have to uncomment it, ex: remove the "--" at the start of the "elseif (Contro........" since it's probably commented out in the public version.

**Comment by billabongbruno** [comment]
> 

Hey there.

First of all, thank you for your help, input and attempts at solving this issue.

I do realize it's a TWOM issue and will follow your advice and open an issue there, to see what the dev can do about it, if anything.

I have no idea on how to launch TWOM:Stories other than the psla:stories redirection, unfortunately.

I am not familiar with LUA coding, but I guess I'll dive into it a bit and see if I can come up with something.
If I am able to boot it at all, even though I'm a novice to LUA and the coding might be rough, I'll be sure to share my findings with you, should anyone have a similar issue, thus allowing HexFlow Custom to have a workaround / fix for all thes multiple-banner apps.

Once again, I cannot thank you enough for the time spent on this.


See you soon.

Best regards,
Bruno.

**Comment by billabongbruno** [comment]
I opened the issue [here](https://github.com/TheOfficialFloW/twom_vita/issues/16) and will keep you posted if and when an answer comes along and if my findings come to fruition.

Thank you once again.


Best regards,
Bruno.

**Comment by BlackSheepBoy69** [comment]
Technically, I wasn't planning to have HexFlow be able to launch the banner, it's just I could make a whole separate vpk (as you saw in the FAILED vpk I sent you) and I could just make it like HexFlow Launcher where it takes an index.lua, but the index.lua for that would be only like 2 lines of code, the system.launch(TWOM:stories)... and system.exit to close the redirector. Ex: The redirector would run on LUA like HexFlow does so it could easily be fixed if there's any further issues... I hope that would be a good solution?

**Comment by billabongbruno** [comment]
> 

I mean, anything works, really. Your solution is perfectly reasonable and seems like a good workaround.

Hopefully Rinne replies soon enough, so that some light can be shed on the matter.

**Comment by billabongbruno** [comment]
> Nope... For whatever reason it seems to refuse to launch directly by a separate bubble. The only hope might be indirectly (with LUA). You can find my direct attempt here:
> 
> https://www.mediafire.com/file/ylga5a2om8fv0zt/TWOM_Stories_Redirector_FAILED.vpk/file
> 
> Note: The apptitle is "Wordle" because I was too lazy to change it. I'm not super passionate about getting it fixed, if you are, please go to the This War Of Mine Github page and open an issue:
> 
> Title: LUA Direct Launch
> 
> Text something like: I was trying to make a little separate bubble that could launch straight to TWOM:Stories (so it could easily be accessed by HexFlow Launcher) using your Lua Player Plus. I tried this:
> 
> ```
> System.executeUri("psgm:play?titleid=TWOM00000&param=psla:stories")
> ```
> 
> but it doesn't seem to work. Do you know what would be able to launch into TWOM:Stories?
> 
> Not part of the text to put in the issue, this is just me talking: I really did try that but it didn't work. For the wordle thing I just made a copy of the TWOM game, stuffed it into a Wordle VPK (since that's the only homebrew I know that primarily uses PSLA: launch. If you want to try out LUA code, just put it in HexFlow Launcher Custom right below the area that says:
> 
> ```
> elseif (Controls.check(pad, SCE_CTRL_SELECT) and not Controls.check(oldpad, SCE_CTRL_SELECT)) then
> ```
> 
> (Example usage: remove the "--" at the start of the "elseif (Contro........", since it's meant for bugtesting and is commented out in the public version of HexFlow Custom, then in the next line put the "System.executeUri("psgm:play?titleid=TW.........." and you can try editting it in some ways to find a way that might work?)
> 
> Closing the issue for now as it has more to do with the weird way TWOM launches and would be more of an issue for their GitHub. If they say anything back, I'll see what I can do

Hey there again.

So, I am away from home right now, so I can't really check in my PSVita.
I just noted something in the VPK you compiled, while trying to come up with a solution or workaround.

Under settings.cfg, the enable_dlc is set to 0, rather than 1.

As far as I can tell, Stories is a DLC itself.
I know that in my "normal" installation, I did change enable_dlc to 1 and while have the TWOM bubble, both banners launched each version of the game.

Could it be possible that this is the issue?

(I have no way of testing now, only in about 15 hours, once I get home).

**Comment by BlackSheepBoy69** [comment]
You definitely aren't being a bother, I have probably spent like an hour max total working on the issue, it's just I was mostly waiting on Rinne to say how to make it work. I tried enabling DLC's, but it still says "Error could not load ux0:data/twom/libAndroidGame.so"

I think it might be related that kubridge does special actions based on the app's ID and this app ID is not the same as TWOM's.

Waiting on Rinne to reply back because I tested all I could think to try
